packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -610,15 +610,43 @@ export class NetworkMonitor {
         )
       }
 
-      // Check for validity with epoch manager currentEpoch, and fetch block hash
-      const validBlock =
-        result.data.network.latestValidBlockNumber.epochNumber ==
-        (await this.currentEpochNumber())
-          ? result.data.network.latestValidBlockNumber
-          : result.data.network.latestValidBlockNumber.previousBlockNumber
-
-      // Resolve block hash for the given block number.
-      // Calls the configured provider for blocks from protocol chain, or Graph Node otherwise.
+      // Check for validity with epoch manager currentEpoch, and fetch block hash after that.
+      const epochManagerCurrentEpoch = await this.currentEpochNumber()
+      const epochSubgraphCurrentEpoch =
+        result.data.network.latestValidBlockNumber.epochNumber
+
+      // The Epoch Manager will always lead new epochs, so it should be minuend.
+      const epochDifference = epochManagerCurrentEpoch - epochSubgraphCurrentEpoch
+
+      let validBlock
+      switch (true) {
+        case epochDifference < 0:
+          // Somehow, the Epoch Manager is behind the Epoch Subgraph. This is a critical failure.
+          const criticalErrorMessage = `The Epoch Manager is behind the Epoch Subgraph`
+          this.logger.critical(criticalErrorMessage, {
+            epochManagerCurrentEpoch,
+            epochSubgraphCurrentEpoch,
+          })
+          throw new Error(criticalErrorMessage)
+        case epochDifference === 0:
+        case epochDifference === 1:
+          // It is acceptable that the Epoch Subgraph stays at least one epoch behind the Epoch
+          // Manager, considering this measurement could have happened during an epoch transition.
+          validBlock = result.data.network.latestValidBlockNumber
+        case epochDifference === 1:
```

Comment:
This case, `epochDifference === 1`, is mentioned twice.  Is that on purpose?

Reply:
Yes, the intention was to avoid a repeated assignment to `validBlock`, which should happen if `epochDistance` is equal to 0 or 1, but to emit a log only if the distance is equal to 1.

Anyhow, I gave up on `switch` in favor of a `if-else` chain which will be more explicit, although also more verbose.